scripts/pre_push_hook.py

```diff
@@ -0,0 +1,184 @@
+deviate from develop.
+(By providing the list of files to `scripts/pre_commit_linter.py`)
+To install the hook manually simply execute this script with the `--install`
+flag.
+"""
+
+import os
+import sys
+import subprocess
+import collections
+import pprint
+import argparse
+import shutil
+
+_PARSER = argparse.ArgumentParser()
+_PARSER.add_argument("remote", nargs="?", help="provided by git before push")
+_PARSER.add_argument("url", nargs="?", help="provided by git before push")
+_PARSER.add_argument("--install", action="store_true", default=False,
+                     help="Install pre_push_hook to the .git/hooks directory")
+_ARGS = _PARSER.parse_args()
+REMOTE = _ARGS.remote
+
+GitRef = collections.namedtuple("GitRef", ["local_ref", "local_sha1",
+                                           "remote_ref", "remote_sha1"])
+
+# git hash of /dev/null, refers to an "empty" commit
+GIT_NULL_COMMIT = "4b825dc642cb6eb9a060e54bf8d69288fbee4904"
+
+
+# caution, __file__ is here *OPPiA/.git/hooks* and not in *OPPIA/scripts*
+FILE_DIR = os.path.abspath(os.path.dirname(__file__))
+OPPIA_DIR = os.path.join(FILE_DIR, os.pardir, os.pardir)
+SCRIPTS_DIR = os.path.join(OPPIA_DIR, "scripts")
+LINTER_SCRIPT = "pre_commit_linter.py"
+LINTER_FILE_FLAG = "--files"
+PYTHON_CMD = "python"
+BACKEND_TEST_SCRIPT = "run_backend_tests.sh"
+FRONTEND_TEST_SCRIPT = "run_frontend_tests.sh"
+
+
+def _start_subprocess_for_result(cmd):
+    """Starts subprocess and returns (stdout, stderr)"""
+    task = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+    out, err = task.communicate()
+    return out, err
+
+
+def _git_diff_names_only(left, right, diff_filter=""):
+    """Compare two branches/commits etc with git.
+    Parameter:
+        left: the lefthand comperator
+        right: the righthand comperator
+        diff_filter: arguments given to --diff-filter (ACMRTD...)
+    Returns:
+        List of file names
+    Raises:
+        ValueError if git command fails
+    """
+    git_cmd = ["git", "diff", "--name-only"]
+    if diff_filter:
+        git_cmd.append("--diff-filter={}".format(diff_filter))
+    git_cmd.extend([left, right])
+    out, err = _start_subprocess_for_result(git_cmd)
+    if not err:
+        return out.splitlines()
+    else:
+        raise ValueError(err)
+
+
+def _compare_to_remote(remote, local_branch, remote_branch=None):
+    """Compare local with remote branch with git diff.
+    Parameter:
+        remote: Git remote being pushed to
+        local_branch: Git branch being pushed to
+        remote_branch: The branch on the remote to test against. If None same
+            as local branch.
+    Returns:
+        List of file names that are modified, changed, renamed or added
+        but not deleted
+    Raises:
+        ValueError if git command fails
+    """
+    remote_branch = remote_branch if remote_branch else local_branch
+    git_remote = "{}/{}".format(remote, remote_branch)
```

Comment:
Actually, could we use '%s/%s' % (..., ...)? We generally use string interpolation by convention, in this codebase.